### PR TITLE
Add form submissions dashboard

### PR DESCRIPTION
**Pull request by Soare-Robert-Daniel**
Closes #.

### Summary

Add form submissions dashboard on the Admin page.

ℹ️ Some discussions with @mghenciu have been made to tweak the aspect.

ℹ️ If the submission has an Email type field, that value will be used instead of Submission ID.

ℹ️ When you change the filter, the page will reload. We are using PHP for getting the posts.

ℹ️ If the user has a Form registered, it will appear in the widget.

### Screenshots 

image

image

https://github.com/Codeinwp/otter-blocks/assets/17597852/63922bde-71e7-4dab-a7be-cc802569092d

----

### Test instructions

⚠️ The widget will appear to all who have at least one widget registered (it does not count if it is still present or not -- practically if the user has used it at least once in the Form). On a fresh WP instance, the widget should not be present.

1. Go to the Admin page and see the Submissions widget.
2. Play with the filter.
3. If you deactivate Otter Pro, you should see the Upsell.

---- 

### Checklist before the final review

- [ ] Included E2E or unit tests for the changes in this PR.
- [x] Visual elements are not affected by independent changes.
- [x] It is at least compatible with the [minimum WordPress version](https://wordpress.org/plugins/otter-blocks/).
- [x] It loads additional script in frontend only if it is required.
- [x] Does not impact the [Core Web Vitals](https://web.dev/vitals/).
- [x] In case of deprecation, old blocks are safely migrated.
- [x] It is usable in Widgets and FSE.
- [x] Copy/Paste is working if the attributes are modified.
- [x] PR is following [the best practices]()

### CODE REVIEW

**Comment by mghenciu** [comment]
Nice job, @Soare-Robert-Daniel .

I did some testing, and here are my findings:
- the upgrade button should open a new Tab, and not the current one 
- the width of the .o-upsell-banner should be **fit-content** with a bottom-margin of 8px. This way the container shouldn't break the widget box. I am open to other suggestions, if any
- there's a typo in the **powerful** word
- we should either use Top/Bottom spacing, or only Gap. Right now we use both of them, and this makes the space between elements - way too large (attached): 
Screenshot 2023-06-22 at 09 17 15

- change the Description to this, so it fits on 2 lines: _With Otter's powerful features, you can easily store and manage form submissions - all in one place._

**Review by HardeepAsrani — CHANGES_REQUESTED** [review]
Apart from the mentioned ones, we can only display the widget if the user is using the Forms Block. We can make use of the options that we save with Forms to make sure they aren't empty.

Let me know if you have any questions.

**Review by HardeepAsrani — APPROVED** [review]
LGTM, thanks for making the changes.

**Comment by rodica-andronache** [comment]
@Soare-Robert-Daniel hey!

- [ ] on a new instance, I've installed both Otter free and pro, and created a new form submission. After that, I deactivated Otter Pro, and I got these notices https://vertis.d.pr/i/uE1vcV.

Also, is visible in the screenshot, that the **Collect your Form Submissions with Otter Blocks** heading is not centered.

- [ ] on an older instance, where I had multiple form submissions, the Date is not showing up in the Dashboard upsell https://vertis.d.pr/i/jVQb5H

File: inc/plugins/class-dashboard.php [hunk 1]
```diff
@@ -266,6 +271,316 @@ private function the_otter_banner() {
...
+	}
+
+	/**
+	 * Display the form submissions widget content.
+	 *
+	 * @return void
+	 */
+	public function form_submissions_widget_content() {
+
+		$is_active = Pro::is_pro_active();
```
Comment (HardeepAsrani) [inline inc/plugins/class-dashboard.php:294]: Thank you Rodica for noticing it, we can just check if the Pro is active or not if it's false then simply return the free code, no point in doing running all the logic that only needs in Pro to do unless we have Pro.

**Comment by Soare-Robert-Daniel** [comment]
> * [ ]  on an older instance, where I had multiple form submissions, the Date is not showing up in the Dashboard upsell https://vertis.d.pr/i/jVQb5H

@rodica-andronache, by older you mean for the first version for Submissions when it was in development? We are interested in only the submissions that were created after the feature was released.

**Comment by rodica-andronache** [comment]
@Soare-Robert-Daniel yes, at that time I've used a site where I had Submissions from when it was in development. I've tried now again on a new site and everything's ok, the date is visible. Thank you!

**Comment by Soare-Robert-Daniel** [comment]
@rodica-andronache, thanks for checking. I put the issue on `Ready To Test` for final approval.

**Comment by rodica-andronache** [comment]
@Soare-Robert-Daniel for me, the title is still not centered. Only if I add `text-align: center;` instead of the code you added, it gets centered, if it helps

**Comment by Soare-Robert-Daniel** [comment]
@rodica-andronache, thanks for the insight. I mistook layout centering with text centering.

**Comment by rodica-andronache** [comment]
@Soare-Robert-Daniel everything's working well 👍 
Just have a small question. If you think it doesn't make sense or it's not something we can tackle now, I will skip this and move the issue to Ready to merge. Let me know.

For filtering the submissions, when you choose something from All/Read/Unread, the page gets refreshed and you go back to the top of the page, which is a bit of a weird behaviour in my opinion. Other widgets seems to be able to handle something similar without scrolling at the top of the page https://vertis.d.pr/v/JZbzc0
Thank you!

**Comment by Soare-Robert-Daniel** [comment]
> For filtering the submissions, when you choose something from All/Read/Unread, the page gets refreshed and you go back to the top of the page, which is a bit of a weird behaviour in my opinion. Other widgets seems to be able to handle something similar without scrolling at the top of the page https://vertis.d.pr/v/JZbzc0 Thank you!

I looked at other widgets on how they do things. They are using AJAX for those updates. The process seemed too complicated for this feature. Not only do you need to pull the data, but also using JS to render it.

An easy and cheap way to do it is to add a hash id in the URL to that it can automatically scroll at the widget: `http://localhost:10007/wp-admin/index.php?otter_form_widget_filter=read&otter_nonce=a4803139c1#otter_form_submissions_widget`

If the users request more on this feature, we can consider making it more powerful. But for now, the trick is good enough since the user might not spend much time on this widget.

**Comment by rodica-andronache** [comment]
@Soare-Robert-Daniel everything's working well, thank you! 👍

**Pull request merged**